IoTuring/Entity/Deployments/Wifi/Wifi.py

```diff
@@ -0,0 +1,321 @@
+                "MCS": r"MCS:\s+(\d+)",
+                "channel": r"channel:\s+([\d,]+)",
+            },
+        }
+        self.platform = OsD.GetOs()
+        if (
+            self.platform == OsD.WINDOWS
+        ):  # Windows shows signal stgrength in percent Unix Systems in dbm
+            self.keySignalStrength = KEY_SIGNAL_STRENGTH_PERCENT
+        else:
+            self.keySignalStrength = KEY_SIGNAL_STRENGTH_DBM
+
+        self.RegisterEntitySensor(
+            EntitySensor(
+                self,
+                self.keySignalStrength,
+                supportsExtraAttributes=True,
+                valueFormatterOptions=ValueFormatterOptions(
+                    ValueFormatterOptions.TYPE_RADIOPOWER
+                ),
+            )
+        )
+
+    def Update(self):
+        p = self.RunCommand(self.commands[self.platform])
+        if not p.stdout:
+            raise Exception("error in GetWirelessInfo\n", p.stderr)
+        wifiInfo = self.GetWirelessInfo(p.stdout)
+        if not wifiInfo:
+            raise Exception("error while parsing wirelessInfo")
+        # set signal strength
+        if "Signal" in wifiInfo:  # Windows
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["Signal"]
+            )
+        elif "Signal_Level" in wifiInfo:  # Linux
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["Signal_Level"]
+            )
+        elif "agrCtlRSSI" in wifiInfo:
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value=wifiInfo["agrCtlRSSI"]
+            )
+        else: # if there is no signal level found the interface might not be connected to an access point
+            self.SetEntitySensorValue(
+                key=self.keySignalStrength, value="not connected"
+            )
+        for key in self.patterns[self.platform]:
+            extraKey = "EXTRA_KEY_" + key.upper().replace(" ", "_")
+            self.SetEntitySensorExtraAttribute(
+                sensorDataKey=self.keySignalStrength,
+                attributeKey=globals()[extraKey],
+                attributeValue=wifiInfo[key] if key in wifiInfo else "not available",
+            )
+
+    def GetWirelessInfo(self, stdout):
+        wifi_info = {}
+        for key, pattern in self.patterns[self.platform].items():
+            match = re.search(pattern, stdout, re.IGNORECASE)
+            if match:
+                wifi_info[key] = match.group(1) if match.group(1) else match.group(0)
+
+        self.Log(self.LOG_DEBUG, wifi_info)
+        return wifi_info
+
+    @classmethod
+    def ConfigurationPreset(cls) -> MenuPreset:         
+
+        preset = MenuPreset()
+        preset.AddEntry(
+            name="Interface to check",
+            key=CONFIG_KEY_WIFI,
+            mandatory=True,
+            question_type="select",
+            choices=NIC_CHOICES
+        )
+        return preset
```

Comment:
shall i stick with the select, even if there is only one wireless interface? If there is only one we can just configure that.
Though i don't really know how i can do that. Setting a Configuration value without the whole preset flow.